vnext/Desktop.UnitTests/MemoryMappedBufferTests.cpp

```diff
@@ -0,0 +1,149 @@
+#include "MemoryMappedBuffer.h"
+#include "utilities.h"
+
+#include <CppUnitTest.h>
+
+#include <shlwapi.h>
+#include <windows.h>
+
+#include <cstring>
+#include <memory>
+
+using facebook::jsi::Buffer;
+using facebook::jsi::JSINativeException;
+using facebook::react::utilities::checkedReinterpretCast;
+using Microsoft::JSI::makeMemoryMappedBuffer;
+using Microsoft::VisualStudio::CppUnitTestFramework::Assert;
+
+namespace {
+
+static constexpr const wchar_t *const testFileName = L"temp.txt";
+
+static void writeTestFile(const char *const content, size_t size) {
+  FILE *testFilePtr;
+
+  // We need pass in the "b" flag to ensure that fwrite modify its input when
+  // outputting to disk.
+  if (_wfopen_s(&testFilePtr, testFileName, L"wb")) {
+    return;
+  }
+
+  std::unique_ptr<FILE, decltype(&fclose)> testFilePtrWrapper(
+      testFilePtr, fclose);
+
+  fwrite(content, sizeof(*content), size, testFilePtrWrapper.get());
+}
+
+static void deleteTestFile() {
+  if (!DeleteFileW(testFileName)) {
+    Assert::IsTrue(GetLastError() == ERROR_FILE_NOT_FOUND);
+  }
+  Assert::IsTrue(!PathFileExistsW(testFileName));
```

Comment:
>Assert::IsTrue [](start = 2, length = 14)

Asserts are used for verifying functionality of our code. Here, you are actually setting up a test, you are not writing a unit test for PathFileExists Api. Therefore, this should be crash or throw exception, instead of Assert.

Reply:
I changed it so that it crashes here instead.

---
In reply to: [320916475](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r320916475) [](ancestors = 320916475)